src/components/Notification/InactivityMessage.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Modal, Button } from '@mui/base';
+
+/**
+ * Inactivity Notification Component - Component that displays a popup modal
+ * after 3 minutes of inactivity, prompting the user to either logout or
+ * continue their session.
+ *
+ * @memberof Notification
+ * @name InactivityMessage
+ */
+
+const InactivityMessage = () => {
+  const navigate = useNavigate();
+
+  const [showPopup, setShowPopup] = useState(false);
+  const [activeUser, setActiveUser] = useState(false);
+
+  // Checks for active user by looking for a loggedIn key in local storage
+  useEffect(() => {
+    const activeCheck = localStorage.getItem('loggedIn');
+    if (activeCheck) setActiveUser(activeCheck);
+  }, []);
+
+  // Toggles the popup after three minutes of inactivity
+  useEffect(() => {
+    let timer = null;
+
+    const resetTimeout = () => {
+      if (timer) clearTimeout(timer);
+  
+      timer = setTimeout(() => {
+        setShowPopup(true);
+      }, 180000)
+    }
+  
+    const handleUserActivity = () => {
+      resetTimeout();
+    };
+  
+    resetTimeout();
+  
+    window.addEventListener('mousedown', handleUserActivity);
+    window.addEventListener('mousemove', handleUserActivity);    
+
+    return () => {
+      clearTimeout(timer);
```

Comment:
I think this would work just fine. How about adding "touchstart" as well to cover for mobile cases and touch screens? I think that'll be great!